blockchain/fullblocktests/error.go

```diff
@@ -0,0 +1,308 @@
+
+	// ErrBadCoinbaseFraudProof indicates that the fraud proof for a coinbase
+	// input was non-null.
+	ErrBadCoinbaseFraudProof = ErrorKind("ErrBadCoinbaseFraudProof")
+
+	// ErrBadCoinbaseAmountIn indicates that the AmountIn (=subsidy) for a
+	// coinbase input was incorrect.
+	ErrBadCoinbaseAmountIn = ErrorKind("ErrBadCoinbaseAmountIn")
+
+	// ErrBadStakebaseAmountIn indicates that the AmountIn (=subsidy) for a
+	// stakebase input was incorrect.
+	ErrBadStakebaseAmountIn = ErrorKind("ErrBadStakebaseAmountIn")
+
+	// ErrBadStakebaseScriptLen indicates the length of the signature script for
+	// a stakebase transaction is not within the valid range.
+	ErrBadStakebaseScriptLen = ErrorKind("ErrBadStakebaseScriptLen")
+
+	// ErrBadStakebaseScrVal indicates the signature script for a stakebase
+	// transaction was not set to the network consensus value.
+	ErrBadStakebaseScrVal = ErrorKind("ErrBadStakebaseScrVal")
+
+	// ErrScriptMalformed indicates a transaction script is malformed in some
+	// way.  For example, it might be longer than the maximum allowed length or
+	// fail to parse.
+	ErrScriptMalformed = ErrorKind("ErrScriptMalformed")
+
+	// ErrScriptValidation indicates the result of executing transaction script
+	// failed.  The error covers any failure when executing scripts such
+	// signature verification failures and execution past the end of the stack.
+	ErrScriptValidation = ErrorKind("ErrScriptValidation")
+
+	// ErrNotEnoughStake indicates that there was for some SStx in a given
+	// block, the given SStx did not have enough stake to meet the network
+	// target.
+	ErrNotEnoughStake = ErrorKind("ErrNotEnoughStake")
+
+	// ErrStakeBelowMinimum indicates that for some SStx in a given block, the
+	// given SStx had an amount of stake below the minimum network target.
+	ErrStakeBelowMinimum = ErrorKind("ErrStakeBelowMinimum")
+
+	// ErrNotEnoughVotes indicates that a block contained less than a majority
+	// of voters.
+	ErrNotEnoughVotes = ErrorKind("ErrNotEnoughVotes")
+
+	// ErrTooManyVotes indicates that a block contained more than the maximum
+	// allowable number of votes.
+	ErrTooManyVotes = ErrorKind("ErrTooManyVotes")
+
+	// ErrFreshStakeMismatch indicates that a block's header contained a
+	// different number of SStx as compared to what was found in the block.
+	ErrFreshStakeMismatch = ErrorKind("ErrFreshStakeMismatch")
+
+	// ErrInvalidEarlyStakeTx indicates that a tx type other than SStx was found
+	// in the stake tx tree before the period when stake validation begins, or
+	// before the stake tx type could possibly be included in the block.
+	ErrInvalidEarlyStakeTx = ErrorKind("ErrInvalidEarlyStakeTx")
+
+	// ErrTicketUnavailable indicates that a vote in the block spent a ticket
+	// that could not be found.
+	ErrTicketUnavailable = ErrorKind("ErrTicketUnavailable")
+
+	// ErrVotesOnWrongBlock indicates that an SSGen voted on a block not the
+	// block's parent, and so was ineligible for inclusion into that block.
```

Comment:
maybe voted on a block that's not the parent?

Reply:
Added.